Sources/Turf/JSON.swift

```diff
@@ -0,0 +1,183 @@
+            return value.rawValue
+        case let .array(value):
+            return value.rawValue
+        }
+    }
+}
+
+/**
+ A JSON array of `JSONValue` instances.
+ */
+public typealias JSONArray = [JSONValue?]
+
+extension JSONArray: RawRepresentable {
+    public typealias RawValue = [Any?]
+    
+    public init(rawValue values: RawValue) {
+        self = values.map(JSONValue.init(rawValue:))
+    }
+    
+    public var rawValue: RawValue {
+        return map { $0?.rawValue }
+    }
+}
+
+/**
+ A JSON object represented in memory by a dictionary with strings as keys and `JSONValue` instances as values.
+ */
+public typealias JSONObject = [String: JSONValue?]
+
+extension JSONObject: RawRepresentable {
+    public typealias RawValue = [String: Any?]
+    
+    public init(rawValue: RawValue) {
+        self = rawValue.mapValues { $0.flatMap(JSONValue.init(rawValue:)) }
+    }
+    
+    public var rawValue: RawValue {
+        return mapValues { $0?.rawValue }
+    }
+}
+
+extension JSONValue: ExpressibleByStringLiteral {
+    public init(stringLiteral value: StringLiteralType) {
+        self = .init(value)
+    }
+}
+
+extension JSONValue: ExpressibleByIntegerLiteral {
+    public init(integerLiteral value: IntegerLiteralType) {
+        self = .init(value)
+    }
+}
+
+extension JSONValue: ExpressibleByFloatLiteral {
+    public init(floatLiteral value: FloatLiteralType) {
+        self = .init(value)
+    }
+}
+
+extension JSONValue: ExpressibleByBooleanLiteral {
+    public init(booleanLiteral value: BooleanLiteralType) {
+        self = .init(value)
+    }
+}
+
+extension JSONValue: ExpressibleByArrayLiteral {
+    public typealias ArrayLiteralElement = JSONValue?
+    
+    public init(arrayLiteral elements: ArrayLiteralElement...) {
+        self = .init(elements)
+    }
+}
+
+extension JSONValue: ExpressibleByDictionaryLiteral {
+    public typealias Key = String
+    public typealias Value = JSONValue?
+    
+    public init(dictionaryLiteral elements: (Key, Value)...) {
+        self = .init(.init(uniqueKeysWithValues: elements))
+    }
+}
+
+extension JSONValue: Codable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if let boolean = try? container.decode(Bool.self) {
+            self = .boolean(boolean)
+        } else if let number = try? container.decode(Double.self) {
+            self = .number(number)
+        } else if let string = try? container.decode(String.self) {
+            self = .string(string)
+        } else if let object = try? container.decode(JSONObject.self) {
+            self = .object(object)
+        } else if let array = try? container.decode(JSONArray.self) {
+            self = .array(array)
+        } else {
+            throw DecodingError.typeMismatch(JSONValue.self, DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Unable to decode as a JSONValue."))
+        }
+    }
```

Comment:
how does this decode `null`?

Reply:
It doesn’t: `null` is represented externally by `Optional.none`.